### PR TITLE
Showing correct version

### LINKED ISSUE #1927: Y24-328 - [BUG] Show correct version labels in each environment
**Describe the bug**
The correct version labels should be shown on the page footers.

See the current labels below:

**Local:**
Screenshot 2024-09-17 at 10 34 10
This is good and does not need to change

**UAT:**
Screenshot 2024-09-17 at 10 35 02
Should not show WIP and should show the version (ie: `v3.58.0`) instead of the deployment time (`20240917...`)

**Production:**
Screenshot 2024-09-17 at 10 35 34
Should not show WIP and should show the version (ie: `v3.58.0`) instead of the deployment time (`20240917...`)

The text in the link (in blue) must not change, so that compatability with integration-suite version tracking is maintained.

**RT Ticket Number**
If applicable

**To Reproduce**
Steps to reproduce the behaviour:
1. Go to Limber in local, uat/staging, and production environments
2. Look at the text in the footer 

**Expected behaviour**
See notes above.

**Additional context**
The text in the link (in blue) must not change, so that compatability with integration-suite version tracking is maintained.

### PR DESCRIPTION
**Pull request by dasunpubudumal**
Closes #1927

#### Changes proposed in this pull request

- Fixes version information in the UI footer

#### Instructions for Reviewers

_[All PRs] - Confirm PR template filled_  
_[Feature Branches] - Review code_  
_[Production Merges to `main`]_  
 &nbsp; &nbsp; \- _Check story numbers included_  
 &nbsp; &nbsp; \- _Check for debug code_  
 &nbsp; &nbsp; \- _Check version_

### CODE REVIEW

File: lib/deployed_version.rb [hunk 1]
```diff
@@ -53,7 +53,7 @@ def version_hash
     end
 
     def version_label
-      version_hash ? "#{major}.#{minor}.#{patch}" : 'WIP'
+      major.to_s == '0' && minor.to_s == '0' && patch.to_s == '0' ? 'WIP' : "#{major}.#{minor}.#{patch}"
```
[the remark this answers is not in the document]
Comment (BenTopping) [reply lib/deployed_version.rb:56]: Tag gets set in the [compile-build script](https://github.com/sanger/sequencescape/blob/develop/compile-build#L11).

[the remark this answers is not in the document]
Comment (dasunpubudumal) [reply lib/deployed_version.rb:56]: Hi Stephen, sorry for replying to this (very) late. Yes indeed it is a problem with the TAG file that's created. The TAG file _is_ written, but without any content in it! I will have a look why. Thanks for pointing it out.

[the remark this answers is not in the document]
Comment (dasunpubudumal) [reply lib/deployed_version.rb]: The `automated_release.yml` workflow somehow doesn't populate the `TAG` file. When we create a release manually, the `append_release.yml` properly sets the `TAG` file. Something fishy going on in the automated release workflow.

**Review by andrewsparkes — APPROVED** [review]

File: compile-build [hunk 2]
```diff
@@ -10,9 +10,9 @@ bundle exec rake assets:clobber
 echo 'Compiling assets'
 bundle exec rake assets:precompile
 echo 'Generating sha'
-git rev-parse HEAD > REVISION
-git tag -l --points-at HEAD --sort -version:refname | head -1 > TAG
-git rev-parse --abbrev-ref HEAD > BRANCH
+echo ${1:-$(git rev-parse HEAD)} > REVISION
+echo ${2:-$(git tag -l --points-at HEAD --sort -version:refname | head -1)} > TAG
+echo ${3:-$(git rev-parse --abbrev-ref HEAD)} > BRANCH
```
[the remark this answers is not in the document]
Comment (dasunpubudumal) [reply compile-build:15]: It takes arguments provided into the script and if the argument is absent it evaluates the expression. This is the same as sequencescape.

**Pull request merged**